tools/imas/imas_wrapper.py

```diff
@@ -0,0 +1,542 @@
+	eq_psi = eq_psi - eq_psi[0]	
+
+	# load the coreprof psi
+	if len(coreprof.profiles_1d[0].grid.psi) == 0:
+		logging.warning('The poloidal flux from core profiles IDS seems to be empty!\n')
+		raise Exception('Error in loading core profiles data. Please check the input data!')
+	else:
+		cp_psi = coreprof.profiles_1d[0].grid.psi
+		if log:
+			logging.info('The poloidal flux from core profiles IDS loaded successfully!\n')
+	
+	# interpolate the coreprof rho from the coreprof psi to the equilibrium psi
+	interp_radius = CubicSpline(-cp_psi, cp_radius)
+	radius = interp_radius(-eq_psi)
+	
+	# Check if the interpolation gave a correct answer
+	error = (radius[-1] - cp_radius[-1]) / radius[-1]
+	if abs(error) > 0.1:
+		raise Exception("The interpolation gave significant difference in the equilibrium and coreprofile radii. Cannot continue, please check the input parameters.")
+	
+	#load wall data with error handling
+	if wall.description_2d[0].vessel.unit[0].annular.resistivity == -9e+40:
+		logging.warning('The the wall resistivity from wall IDS seems to be empty!\n')
+		raise Exception('Error in loading wall data. Please check the input data!')
+	else:
+		resistivity = wall.description_2d[0].vessel.unit[0].annular.resistivity
+		if log:
+			logging.info('The major radius of the magnetic axis from core profiles IDS loaded successfully!\n')
+
+	if setUpDream:
+		
+		ds = DREAMSettings()
+		
+		ds.radialgrid.setCustomGridPoints(radius)
+
+		if wall_radius == -999:
+			raise Exception('If you want to set up a dream settings object, please specify the wall radius!')
+		else:
+			ds.radialgrid.setWallRadius(wall_radius)
+			
+		ds.radialgrid.setMajorRadius(R)
+		ds.radialgrid.setB0(abs(B0))
+		
+		for i in range(len(coreprof.profiles_1d[0].ion)):
+			ds.eqsys.n_i.addIon(ion_names[i], Z=ion_charges[i], iontype=Ions.IONS_DYNAMIC, Z0=ion_charges[i], n=ion_densities[i], r=cp_radius)
+			
+		ds.eqsys.T_cold.setInitialProfile(T_cold, radius=cp_radius)
+		ds.eqsys.T_cold.setType(Tcold.TYPE_SELFCONSISTENT)
+		
+		if e_field!=-999:
+			ds.eqsys.E_field.setType(ElectricField.TYPE_SELFCONSISTENT)
+			ds.eqsys.E_field.setInitialProfile(efield=e_field, radius=cp_radius)
+		
+		else:
+			inverse_wall_time = resistivity / (4*np.pi*1e-7*R*np.log(R/wall_radius))
+			if log:
+				logging.info('The iverse wall time is %s 1/s with the given wall radius.\n', inverse_wall_time)
+		
+			ds.eqsys.E_field.setType(ElectricField.TYPE_SELFCONSISTENT)
+			ds.eqsys.E_field.setBoundaryCondition(ElectricField.BC_TYPE_SELFCONSISTENT, inverse_wall_time=inverse_wall_time, R0=R)
+		
+		ds.eqsys.j_ohm.setInitialProfile(j_tor, radius=eq_radius, Ip0=Ip)
+		
+		return ds
+	
+	else:
+		print('return a list? hdf5?')
+		return 'TO DO'
+	
+	
+def readInIDS(shot, run, tokamak, user=os.getlogin(), log=False, setUpDream=True, wall_radius=-999):
+	
+	# TODO full IDS read in
+	
+	# open shot file
+	dataEntry = imas.DBEntry(imasdef.MDSPLUS_BACKEND, tokamak, int(shot), int(run), user_name=user)
```

Comment:
Which backends are available? If you have to specify the backend (which seems very weird to me, considering that the purpose of IMAS was to not use things like MDSplus) I guess it should be provided as an input parameter by the user?

Reply:
This is a bit outside my expertise, but if I recall correctly, the IDS structure can be defined in multiple backends, inluding MDSplus and HDF5 (I am not sure if there are any other). It is a remnant of old implementation of the CPOs, when they considered using other stuff than MDSplus, but they sticked with it. As far as I am aware only the MDSPLUS_BACKEND works with IMAS, and it has to be specified. The DBEntry function is not something I wrote, I am not sure this field can be circumvented.  We could ask Dmitriy Yadikin, if you want to do something about this.

Reply:
Ok, if MDSplus is the only actual working backend then this is probably fine. Possibly we might want to add the backend as an option to ``readInIDSSlice()``, with this as default value? (I also noticed now that this is specified in at least one more place in this function)

Reply:
I didn't want to add it as an option to calling the functions as currently it does not take any other value as a valid option and I do not believe there are any plans to have another backend in the near future. I can add it if you need it, but I think it could be confusing, and would not have any purpose. It should be rather simple to add it if another backend is introduced.
Also, I am not sure what you meant by the sentence in the brackets. I did not find another place where it was specified in `readInIDSSlice()`, only in `readInIDS()`. Can you let me know in what line did you find it?

Reply:
Ok, but then it sounds like maybe the backend is not needed as an input parameter for now.

(The lines where the MDSplus backend is referenced are 23 and 259)